AirCastingTests/SearchAndFollowTests/AirBeamMeasurementDownloaderTests.swift

```diff
@@ -0,0 +1,49 @@
+// Created by Lunar on 03/08/2022.
+//
+import XCTest
+import Resolver
+@testable import AirCasting
+
+class AirBeamMeasurementDownloaderTests: ACTestCase {
+    lazy var sut = AirBeamMeasurementsDownloaderDefault()
+    let clientSpy = APIClientSpy()
+    
+    func test_whenRequestingDataFromServer_shouldUseAssertedPath() {
+        Resolver.test.register { self.clientSpy as APIClient }
+        sut.downloadStreams(with: Int.default, completion: { _ in })
+        guard let url = URLComponents(string: clientSpy.request.description)?.url else { XCTFail("Problem when trying to get url."); return }
+        XCTAssertEqual(url.path, "/api/fixed/sessions/\(Int.default)/streams.json")
+    }
+    
+    func test_whenRequestingDataFromServer_shouldUseGetMethod() {
+        Resolver.test.register { self.clientSpy as APIClient }
+        sut.downloadStreams(with: Int.default, completion: { _ in })
+        XCTAssertEqual(clientSpy.request.httpMethod, "GET")
+    }
+    
+    func test_whenRequestingDataFromServer_shouldUseGivenSesionUUIDinsidePath() {
+        Resolver.test.register { self.clientSpy as APIClient }
+        let sessionUUID = 123456-789
+        sut.downloadStreams(with: sessionUUID, completion: { _ in })
+        guard let url = URLComponents(string: clientSpy.request.description)?.url else { XCTFail("Problem when trying to get url."); return }
+        XCTAssertEqual(url.path, "/api/fixed/sessions/\(sessionUUID)/streams.json")
+    }
+    
+    class APIClientSpy: APIClient {
+        
+        struct URLSessionCancellable: Cancellable {
+            weak var dataTask: URLSessionDataTask?
+            
+            func cancel() {
+                dataTask?.cancel()
+            }
+        }
+        
+        var request: URLRequest!
```

Comment:
1. This should be an optional because what happens if our `sut` doesn't call `requestTask` at all? (tests would crash). 
2. I also like more verbose naming in tests. Maybe `capturedRequest`? It will make test-site much more readable
3. What if the `sut` calls `requestTask` multiple times by mistake? We won't get a failing test. This is why a good practice in `Spy` objects is to make capturing variables an arrays. Like that: `var capturedRequests: [URLRequest]`. If you do so you won't need to make it an optional too.

Reply:
Thanks so much for this step-by-step explanation - really helps a lot 💙